CompilerSource/parser/collect_variables.cpp

```diff
@@ -108,6 +108,7 @@ class DeclGatheringVisitor : public AST::Visitor {
       if (left->name.content == "global" && node.operation.type == enigma::parsing::TokenType::TT_DOT) {
         auto right = node.right->As<AST::IdentifierAccess>();
         parsed_scope->globals[right->name.content] = dectrip("var");
+        parsed_scope->locals[right->name.content] = dectrip("var");
```

Comment:
This way, we can have both versions for any declared variable: a local one accessed through `foo` directly, and a global one accessed through `global.foo`.